test/unittest/tests/helpers/vector_bool.cpp

```diff
@@ -0,0 +1,139 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *      All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "vector_bool.h"
+#include <algorithm>
+#include <charconv>
+#include <iostream>
+
+std::vector<bool>
+vector_bool_from_hex_string(std::string const &hex_str)
+{
+    std::vector<bool> result;
+
+    for (auto iter = hex_str.crbegin(); iter != hex_str.crend(); ++iter) {
+        int value_of_hex_char = std::stoi(std::string(1, *iter), nullptr, 16);
+        result.push_back((value_of_hex_char & 1) != 0);
```

Comment:
The loop performs a reverse iteration through the string so that bits are appended (rather than prepended) to the bitmap. I've added a comment to that effect, and hopefully it's now clearer.

I've also added some extra test cases to confirm that 0x3c is encoded as 0011 1100 (note the bitmap initialisers are least significant bit first).

```
    test_hex_conversion({false, false, false, false, true, false, false, false}, "10");
    test_hex_conversion({true, false, false, false, false, false, false, false}, "01");
    test_hex_conversion({false, false, true, true, true, true, false, false}, "3c");
    test_hex_conversion({true, true, false, false, false, false, true, true}, "c3");
```

Reply:
Hi Jeremy, I'm sorry but I still don't understand why you're reversing the bits, according to your tests quoted above:
```
0x10 == 0b00001000
0x01 == 0b10000000
```
But my expectation is:
```
0x10 == 0b00001000
0x01 == 0b00000001
```
Here is a demonstration in c++17:
```
ip-10-0-101-50:tmp [1]$ cat binary.cc
// clang -std=c++17 -o binary binary.cc
#include <iostream>
#include <iomanip>
int main(void) {
	using namespace std;
	unsigned char b1 = 0b00010000;
	unsigned char b2 = 0b00000001;
	cout << setw(2) << hex <<
		"b1 = " << unsigned(b1) << endl <<
		"b2 = " << unsigned(b2) << endl;
	return 0;
}

$ ./binary
b1 = 10
b2 = 1
```